examples/direct_fidelity_estimation.py

```diff
@@ -95,6 +96,22 @@ def compute_characteristic_function(circuit: cirq.Circuit,
     return sigma_i
 
 
+@dataclass
+class PauliTrace:
+    """
+    A class that contains the Pauli states as described on page 2 of:
+    https://arxiv.org/abs/1104.3835
+    """
+    # Pauli string.
+    P_i: cirq.PauliString
+    # The pure state for the Pauli string above, formally defined at bottom of
```

Comment:
Wait, this isn't true though, is it? Rho with the caret is the ideal pure state. Rho with the index i is the trace of the product of rho caret with ith Pauli string. You can also see this from the type annotation: rho_i is a float, so it can't be a pure state.

I see three ways of defining this well:

1. Experimentalist: "Average value of the ith Pauli observable in the ideal pure state."
2. Theoretician: "Coefficient of the ideal pure state expanded in the Pauli basis scaled by sqrt(dim H)".
3. Mathematician: "Hilbert-Schmidt inner product of the ideal pure state and the ith Pauli basis element."
4. Just show me the math: Simply state the formula.

My favorite is actually 2 since it hints a little bit at how DFE works behind the scenes. Your choice though.

Reply:
Done